catalog/transformer/catalog-transformer-rtf/src/main/java/ddf/catalog/transformer/output/rtf/model/ExportCategory.java

```diff
@@ -0,0 +1,138 @@
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.output.rtf.model;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.types.Core;
+import java.util.AbstractMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang.WordUtils;
+
+public class ExportCategory implements RtfCategory {
+
+  private String title;
+  private List<String> attributes;
+
+  public abstract class ExportValue<T, R extends ValueType> {
+    public abstract T getValue();
+
+    public abstract R getType();
+  }
+
+  public enum ValueType {
+    SIMPLE,
+    MEDIA,
+    EMPTY
+  }
+
+  class JustValue<T> extends ExportValue {
+
+    private final T value;
+    private final ValueType valueType;
+
+    public JustValue(T value, ValueType type) {
+      this.value = value;
+      this.valueType = type;
+    }
+
+    @Override
+    public T getValue() {
+      return this.value;
+    }
+
+    @Override
+    public ValueType getType() {
+      return this.valueType;
+    }
+  }
+
+  public ExportCategory() {}
+
+  public void init() {
+    // Thanks blueprint!
+  }
+
+  public void destroy(int code) {
+    // Thanks blueprint!
+  }
+
+  public String getTitle() {
+    return title;
+  }
+
+  public void setTitle(String title) {
+    this.title = title;
+  }
+
+  public List<String> getAttributes() {
+    return attributes;
+  }
+
+  public void setAttributes(List<String> attributes) {
+    this.attributes = attributes;
+  }
+
+  public Map<String, ExportValue> toExportMap(Metacard metacard) {
+    return attributes
+        .stream()
+        .map(
+            key ->
+                new AbstractMap.SimpleEntry<>(
+                    attributeKeyFrom(key),
+                    attributeExportValueFrom(key, metacard.getAttribute(key))))
+        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+  }
+
+  private String attributeKeyFrom(String key) {
+    String formattedAttribute =
+        Stream.of(key.split("\\."))
+            .map(part -> part.replaceAll("-", " "))
+            .collect(Collectors.joining(" "));
+
+    return WordUtils.capitalize(formattedAttribute);
+  }
+
+  private ExportValue attributeExportValueFrom(String attributeKey, Attribute attribute) {
+    if (attributeKey == null || attribute == null) {
+      return emptyValue();
+    }
+
+    if (attributeKey.equals(Core.THUMBNAIL)) {
```

Comment:
I'm not sure I understand. Right now this will only handle thumbnails and everything else will be either empty or represented via `toString()` so I don't see why we would care about binary attributes.

Reply:
I'm basically wondering if can simply call `toString()` on an attribute of type BINARY or OBJECT that's not the thumbnail. Can we guarantee that it will not mess up the RTF output? Should we simply output something like the string "Binary" or "Object" in those cases?

Reply:
Since we are only transforming attributes that are explicitly configured by users via category config files they have to have knowledge of the type of data that will be exported into RTF (meaning very little likelihood they add a binary attribute). My view is that we should leave it as is.